src/cmd/ksh93/sh/jobs.c

```diff
@@ -348,6 +348,7 @@ int job_reap(register int sig)
 	Cojob_t		*cjp;
 	int		cojobs;
 	long		cotimeout = sig?0:-1;
+	int was_ttywait_on = sh_isstate(SH_TTYWAIT); /* save tty wait state */
```

Comment:
This moves it within `#if SHOPT_COSHELL`...`#endif`, which I don't think is right.

I was thinking more along the lines of  a`int was_ttywait_on;` declaration on line 347, then actually save the state on line 377... that is the practice in most of the rest of the code.